core/src/modules/ops/profile.ts

```diff
@@ -671,24 +671,43 @@ export namespace ProfileOps {
       (await plugin.readSetting("core", "runs-profile-batch-size")).value
     );
 
-    const profiles: Profile[] = await api.sequelize.query(
-      `
-      SELECT "id" FROM "profiles" WHERE "state"='ready' 
-        AND 0 = (
-          SELECT count("exports"."id") FROM "exports" 
-            WHERE "exports"."profileId"="profiles"."id" 
-            AND "exports"."state" IN ('pending', 'processing')
-        ) AND "id" NOT IN (
-          SELECT DISTINCT("profileId") FROM "profileProperties" 
-          JOIN properties ON "properties"."id"="profileProperties"."propertyId" 
-          WHERE "properties"."directlyMapped"=true AND "rawValue" IS NOT NULL
-        ) LIMIT ${limit};
-      `,
-      {
-        model: Profile,
-      }
+    let profiles: Profile[] = [];
+    const directlyMappedProperties = (await Property.findAllWithCache()).filter(
+      (p) => p.directlyMapped
     );
 
+    if (directlyMappedProperties.length === 0) {
+      // We have no directly mapped Property and every profile should be removed
+      // It's safe to assume that if there are no Properties, we aren't exporting
+      profiles = await Profile.findAll({
+        attributes: ["id"],
+        where: { state: "ready" },
+        limit,
+      });
+    } else {
+      // We have directly mapped Properties and we only want to remove those Profiles with a null "user_id" (directlyMapped) Property (and are ready with no exports)
+      profiles = await api.sequelize.query(
+        `
+  SELECT "id" FROM "profiles"
+  WHERE "state"='ready'
+  AND 0 = (
+    SELECT count("exports"."id") FROM "exports"
+      WHERE "exports"."profileId"="profiles"."id"
+      AND "exports"."state" IN ('pending', 'processing')
+  )
+  AND "id" NOT IN (
```

Comment:
Something interesting about this is that it also worked for when we eventually add more directlyMapped properties for different profile types. If any directlyMapped property has a value, it wouldn't be sweeped, but in the new inverted version if any is null, it would be sweeped.

I think this is fine and not really relevant until we do implement profile/entity types, so not something to worry about now. Just noting that we may need to revisit this later.

Reply:
Good point! I think when we have Entity/Models, we'll have to do this in a loop - once per entity/model.  Not all Directly Mapper properties will belong to each type.  The good news is that we'll hopefully get to a point where each type only has one "root" property - the one and only one directly mapped property.